### PR TITLE
Hiddenfiles

### PR DESCRIPTION
**Pull request by kwsp**
This PR addresses https://github.com/filebrowser/filebrowser/issues/1084 through the following:

Frontend:
* Added a checkbox in Profile Settings to toggle show/hide dotfiles at the user level
* When the files api is called on a directory and `showHidden` is `true`, all directory entries that begin with a `.` are prune from the api results.

Backend:
* Added `showHidden` to the `userDefaults` and `userInfo`
* This attribute defaults to false (by default users do not see hidden files) and is sent along with other `userInfo` from the auth endpoint to the frontend. 

@o1egl please review :)

### CODE REVIEW

File: frontend/src/views/Files.vue [hunk 1]
```diff
@@ -9,6 +9,7 @@
         <span class="chevron"><i class="material-icons">keyboard_arrow_right</i></span>
         <router-link :to="link.url">{{ link.name }}</router-link>
       </span>
+
```
Comment (o1egl) [inline frontend/src/views/Files.vue]: Why did you add a newline here?

File: frontend/src/views/Files.vue [hunk 2]
```diff
@@ -145,12 +167,16 @@ export default {
...
 
       try {
-        const res = await api.fetch(url)
+        let res = await api.fetch(url)
 
         if (clean(res.path) !== clean(`/${this.$route.params.pathMatch}`)) {
           return
         }
 
+        if (res.isDir && !this.user.showHidden) {
```
Comment (o1egl) [inline frontend/src/views/Files.vue:175]: why is it applied only for dirs?

**Review by o1egl — COMMENTED** [review]
Great job. But I think this property should be active by default.

File: frontend/src/views/Files.vue (hunk 1 above)
Comment (kwsp) [reply frontend/src/views/Files.vue]: I think I added some code here in a previous commit and removed it as I changed the approach and accidentally left a new line. Could remove it if you want.

File: frontend/src/views/Files.vue (hunk 2 above)
Comment (kwsp) [reply frontend/src/views/Files.vue:175]: The frontend only shows the 'file browser' window when the result is a directory. If the result is a file, the previous of the file is shown, so the logic to hide dotfiles in the `browser` view doesn't apply.

**Comment by kwsp** [comment]
> Great job. But I think this property should be active by default.

I've just made 'show hidden files' the default.

**Comment by o1egl** [comment]
1. In current implementation showHidden will be true only for new users. We can reverse this flag to `hideHidden` or `hideDotFiles`. 
2. I don't think that this type filter should be done on client side. Let's move it to the server side.

**Comment by ramiresviana** [comment]
Filtering do not work on the result of files.fetch(), it causes wrong values for items indexes.

For me, the checkbox input feels out of place on the file listing, should be self-describing without text labels.

Since is not an essential feature that needs to be always visible on the listing, i think it should be moved to settings in rules section.

**Comment by kwsp** [comment]
> Filtering do not work on the result of files.fetch(), it causes wrong values for items indexes.
> 
> For me, the checkbox input feels out of place on the file listing, should be self-describing without text labels.
> 
> Since is not an essential feature that needs to be always visible on the listing, i think it should be moved to settings in rules section.

I have not tested `files.fetch()`, will look into it.

I've moved the checkbox to profile settings in the most recent commits, please pull those and see. If you have interest in moving the implementation from the client side to the server side as @o1egl suggested, please feel free to take a look! I might not have time in the coming weeks to do this.
